portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -45,23 +56,127 @@ public void tearDown() {
+    int NUM_COMMENTS = 3;
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+    HttpServletRequest getRequest = mock(HttpServletRequest.class);
+    HttpServletResponse getResponse = mock(HttpServletResponse.class);
 
-    when(requestPost.getParameter("text-input")).thenReturn("Test Comment #1");
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);
+
+    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #1");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #2");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #3");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(getRequest.getParameter("max-num")).thenReturn("3");
+    new DataServlet().doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yet
+    assertThat(stringWriter.toString()).contains("Test Comment #1");
+    assertThat(stringWriter.toString()).contains("Test Comment #2");
+    assertThat(stringWriter.toString()).contains("Test Comment #3");
+  }
+
+  @Test
+  public void testGETEmptyDataStore() throws Exception {
+    HttpServletRequest getRequest = mock(HttpServletRequest.class);
+    HttpServletResponse getResponse = mock(HttpServletResponse.class);
 
     StringWriter stringWriter = new StringWriter();
-    PrintWriter writer = new PrintWriter(stringWriter);
-    when(responseGet.getWriter()).thenReturn(writer);
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);  
 
-    new DataServlet().doPost(requestPost, responsePost);
-    new DataServlet().doGet(requestGet, responseGet);
+    new DataServlet().doGet(getRequest, getResponse); 
 
-    writer.flush(); // it may not have been flushed yet...
-    Assert.assertTrue(stringWriter.toString().contains("Test Comment #1"));
+    printWriter.flush(); //may not have flushed yets
+    assertThat(stringWriter.toString().trim()).isEqualTo("[]");
   }
+
+//   @Test 
+//   public void testGETContentType() throws Exception {
+//     HttpServletRequest getRequest = mock(HttpServletRequest.class);
+//     HttpServletResponse getResponse = mock(HttpServletResponse.class);
+
+//     StringWriter stringWriter = new StringWriter();
+//     PrintWriter printWriter = new PrintWriter(stringWriter);
+      
+//     new DataServlet().doGet(getRequest, getResponse); 
+//     when(getResponse.getWriter()).thenReturn(printWriter);
+//     printWriter.flush(); //may not have flushed yets
+//     String responseContentType = getResponse.getContentType();
+//     assertThat(responseContentType).isEqualTo("application/json");  
```

Comment:
Try using Mockitos [verification](https://javadoc.io/static/org.mockito/mockito-core/3.4.4/org/mockito/Mockito.html#verification) for this test.

Reply:
ah I need help on this front...not sure how to use verifications

Reply:
resolved via 1:1